internal-ui/src/components/DashboardComponent.tsx

```diff
@@ -1,23 +1,76 @@
-import React from 'react';
-import { CalendarMonth } from '@patternfly/react-core';
-import { Checkbox } from '@patternfly/react-core';
+import React, { useEffect, useState } from 'react';
 import {
+  CalendarMonth,
+  Checkbox,
   Card,
-  CardBody,
-  CardTitle,
-  CardHeader,
+  Grid,
+  GridItem,
 } from '@patternfly/react-core';
-import { useState } from 'react';
-
+import {
+  ChartDonut,
+  ChartPie,
+  ChartArea,
+  ChartScatter,
+  ChartThemeColor,
+  ChartVoronoiContainer,
+  ChartBar
+} from '@patternfly/react-charts';
+import { getMessagesInRange } from '../services/messages';
+import { Message } from '../Types';
 
-export const SelectFilters = () => {
+export const DashboardComponent = () => {
   const [startDate, setStartDate] = useState<Date | null>(null);
   const [endDate, setEndDate] = useState<Date | null>(null);
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [uniqueSessionsCount, setUniqueSessionsCount] = useState(10);
+  const [uniqueSenders, setUniqueSenders] =useState(10);
+  const [totalSessions, setTotalSessions] = useState(60);
+  const [averageFlows, setAverageFlows] = useState(1000);
+  const [thumbsUp, setThumbsUpCount] = useState(50);
+  const [thumbsDown, setThumbsDownCount] = useState(40);
+  const [toggleState, setToggleState] = useState({
+    showAll: true,  
+    showInternal: false,
+    showExternal: false
+  });
+
+  useEffect(() => {
+    if (startDate && endDate) {
+      const fetchMessages = async () => {
+        const sessionMessages = await getMessagesInRange(undefined, Math.floor(startDate.getTime() / 1000), Math.floor(endDate.getTime() / 1000));
+        setMessages(sessionMessages);
+      };
+      fetchMessages();
+    }
+    setTotalSessions(messages.length)
```

Comment:
What is the format of the getMessagesInRange function again? I thought it returned the messages in terms of their sessions, if not I'll modify that later.

If it's not sessions here, then the totalsessions wouldnt be the messages.length